cargo-dist/src/lib.rs

```diff
@@ -57,8 +57,88 @@ pub mod tasks;
 #[cfg(test)]
 mod tests;
 
+/// dist env test -- make sure we have everything we need for a build.
+pub fn do_env_test(cfg: &Config) -> DistResult<()> {
+    let (dist, _manifest) = tasks::gather_work(cfg)?;
+
+    let local_builds = cfg.artifact_mode == ArtifactMode::Local;
```

Comment:
We've got two more artifact modes to check for too - `ArtifactMode::Host` (build whatever the current host can build, including local and global builds) and `ArtifactMode::All` (build everything).

Reply:
Changed to checking for global builds and changed `&& local_builds` to `&& !global_builds`, since that seems like what's actually wanted here.